client/src/js/references/components/__tests__/Clone.test.js

```diff
@@ -0,0 +1,170 @@
+                value: "bar",
+                error: false
+            }
+        };
+    });
+    it("should render", () => {
+        const wrapper = shallow(<CloneReference {...props} />);
+        expect(wrapper.state()).toMatchSnapshot();
+    });
+
+    it("handleChange() should return setState() when [name===name]", () => {
+        e.target = {
+            name: "name",
+            value: "foo",
+            error: "error"
+        };
+        const wrapper = shallow(<CloneReference {...props} />);
+        wrapper.find("ReferenceForm").simulate("change", e);
+        expect(wrapper.state()).toEqual({
+            ...state,
+            name: "foo"
+        });
+    });
+    it("handleChange() should return setState() when [name===reference]", () => {
+        e.target = {
+            name: "reference",
+            value: "bar",
+            error: "error"
+        };
+        const wrapper = shallow(<CloneReference {...props} />);
+        wrapper.find("ReferenceForm").simulate("change", e);
+        expect(wrapper.state()).toEqual({
+            ...state,
+            reference: "bar",
+            errorSelect: ""
+        });
+    });
+    it("handleChange() should return setState() when [name!=reference] and [name!=name]", () => {
+        e.target = {
+            name: "foo",
+            value: "bar",
+            error: "baz"
+        };
+        const wrapper = shallow(<CloneReference {...props} />);
+        wrapper.find("ReferenceForm").simulate("change", e);
+        expect(wrapper.state()).toEqual({
+            ...state,
+            foo: "bar"
+        });
+    });
+
+    it("should call handleSelect() when ReferenceSelect is selected", () => {
+        const wrapper = shallow(<CloneReference {...props} />);
+        wrapper.find("ReferenceSelect").simulate("select");
+        expect(wrapper.state()).toEqual({
+            ...state
+        });
+    });
+
+    it("handleSubmit() should return errorName when [this.state.name.length=0]", () => {
+        const wrapper = shallow(<CloneReference {...props} />);
+        wrapper.setState({
+            ...state,
+            name: []
+        });
+        wrapper.find("form").simulate("submit", e);
+        expect(wrapper.state()).toEqual({
+            ...state,
+            name: [],
+            errorName: "Required Field"
+        });
+    });
+    it("handleSubmit() should return errorName when [this.state.name.length=0]", () => {
+        const wrapper = shallow(<CloneReference {...props} />);
+        wrapper.setState({
+            ...state,
+            name: [{ foo: "bar" }],
+            dataType: []
+        });
+        wrapper.find("form").simulate("submit", e);
+        expect(wrapper.state()).toEqual({
+            ...state,
+            name: [{ foo: "bar" }],
+            dataType: [],
+            errorDataType: "Required Field"
+        });
+    });
+    it("handleSubmit() should return errorName when [this.state.name.length=0]", () => {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Test title is used multiple times in the same describe block. (jest/no-identical-title)](https://app.codacy.com/app/virtool/virtool/pullRequest?prid=4644906)